web_sdk/BUILD.gn

```diff
@@ -244,3 +249,160 @@ prebuilt_dart_action("flutter_dartdevc_canvaskit_kernel_sdk") {
+    "file:///" + rebase_path("//third_party/dart/.packages"),
+    "--multi-root-scheme",
+    "org-dartlang-sdk",
+    "--multi-root",
+    "file:///" + rebase_path("$root_out_dir"),
+    "--multi-root-output-path",
+    rebase_path("$root_out_dir/"),
+    "--libraries-file",
+    "org-dartlang-sdk:///flutter_web_sdk/libraries.json",
+    "--inline-source-map",
+    "-DFLUTTER_WEB_USE_SKIA=true",
+    "--modules",
+    "amd",
+    "-o",
+    rebase_path(
+        "$root_out_dir/flutter_web_sdk/kernel/amd-canvaskit-strong/dart_sdk.js"),
+    "--modules",
+    "legacy",
+    "-o",
+    rebase_path(
+        "$root_out_dir/flutter_web_sdk/kernel/legacy-canvaskit-strong/dart_sdk.js"),
+    rebase_path(sdk_dill_strong),
+  ]
+}
+
+# Compiles the DDC SDK's JS code for null safety.
+prebuilt_dart_action("flutter_dartdevc_kernel_sdk_strong") {
+  deps = [
+    ":flutter_dartdevc_kernel_sdk_outline_strong",
+    "//third_party/dart:create_sdk",
+    "//third_party/dart/pkg:pkg_files_stamp",
+    "//third_party/dart/utils/dartdevc:dartdevc_files_stamp",
+    "//third_party/dart/utils/dartdevc:dartdevc_sdk_patch_stamp",
+  ]
+
+  inputs = [ "sdk_rewriter.dart" ] + web_ui_sources + web_engine_sources
+
+  packages = "//third_party/dart/.packages"
+
+  script = "//third_party/dart/pkg/dev_compiler/bin/dartdevc.dart"
+
+  outputs = [
+    "$root_out_dir/flutter_web_sdk/kernel/amd-strong/dart_sdk.js",
+    "$root_out_dir/flutter_web_sdk/kernel/amd-strong/dart_sdk.js.map",
+  ]
+
+  args = [
+    "--enable-experiment=non-nullable",
+    "--sound-null-safety",
+    "-k",
+    "--compile-sdk",
+    "dart:core",
+
+    # Additional Flutter web dart libraries
+    "dart:ui",
+    "dart:_engine",
+    "--no-summarize",
+    "--packages",
+    "file:///" + rebase_path("//third_party/dart/.packages"),
+    "--multi-root-scheme",
+    "org-dartlang-sdk",
+    "--multi-root",
+    "file:///" + rebase_path("$root_out_dir"),
+    "--multi-root-output-path",
+    rebase_path("$root_out_dir/"),
+    "--libraries-file",
+    "org-dartlang-sdk:///flutter_web_sdk/libraries.json",
+    "--inline-source-map",
+    "--modules",
+    "amd",
+    "-o",
+    rebase_path("$root_out_dir/flutter_web_sdk/kernel/amd-strong/dart_sdk.js"),
+    "--modules",
+    "legacy",
+    "-o",
+    rebase_path(
+        "$root_out_dir/flutter_web_sdk/kernel/legacy-strong/dart_sdk.js"),
+  ]
+}
+
+# Compile the DDC SDK's summary.
+prebuilt_dart_action("flutter_dartdevc_kernel_sdk_outline_strong") {
+  deps = [
+    "//third_party/dart:create_sdk",
+    "//third_party/dart/pkg:pkg_files_stamp",
+    "//third_party/dart/utils/dartdevc:dartdevc_files_stamp",
+    "//third_party/dart/utils/dartdevc:dartdevc_sdk_patch_stamp",
+  ]
+
+  inputs = [ "sdk_rewriter.dart" ] + web_ui_sources + web_engine_sources
+
+  outputs = [
+    sdk_dill_strong,
+  ]
+
+  script = "//third_party/dart/utils/bazel/kernel_worker.dart"
+
+  args = [
+    "--enable-experiment=non-nullable",
+    "--sound-null-safety ",
```

Comment:
I'm not familiar with the kernel_worker.dart script and I'm not using it when I build the .dill files in the sdk build target. Does it already support the `--sound-null-safety` flag? I didn't see it in the args in a quick search.

Reply:
I thought It worked locally but I might have missed some flags. At any rate, everything still fails when compiling, I'll double check what dart_sdk is doing and update though

Reply:
everything seems to work OK